commit 2544274a8b8a27f4ea6c457d2c4c32eb1e4cd336
Author: Jia Tan <jiat0218@gmail.com>
Date:   Mon Aug 28 21:54:41 2023 +0800

    Tests: Improve comments in test_index.c.

diff --git a/tests/test_index.c b/tests/test_index.c
index d2c3c0c1..9cf4719b 100644
--- a/tests/test_index.c
+++ b/tests/test_index.c
@@ -141,18 +141,18 @@ test_lzma_index_append(void)
 	assert_lzma_ret(lzma_index_append(idx, NULL, UNPADDED_SIZE_MAX,
 			1), LZMA_DATA_ERROR);
 
-	// Test compressed size growing too large.
+	// Test uncompressed size growing too large.
 	// Should result in LZMA_DATA_ERROR.
 	assert_lzma_ret(lzma_index_append(idx, NULL,
 			UNPADDED_SIZE_MIN, LZMA_VLI_MAX), LZMA_OK);
 	assert_lzma_ret(lzma_index_append(idx, NULL,
 			UNPADDED_SIZE_MIN, 1), LZMA_DATA_ERROR);
 
+	lzma_index_end(idx, NULL);
+
 	// Currently not testing for error case when the size of the Index
 	// grows too large to be stored. This was not practical to test for
 	// since too many Blocks needed to be created to cause this.
-
-	lzma_index_end(idx, NULL);
 }
 
 