commit 6f1790254a03c5edf0f2976f773220f070450acd
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Feb 15 01:53:40 2024 +0800

    Bump version for 5.5.2beta.

diff --git a/src/liblzma/api/lzma/version.h b/src/liblzma/api/lzma/version.h
index 099f42b8..46686dd4 100644
--- a/src/liblzma/api/lzma/version.h
+++ b/src/liblzma/api/lzma/version.h
@@ -22,7 +22,7 @@
 #define LZMA_VERSION_MINOR 5
 
 /** \brief Patch version number of the liblzma release. */
-#define LZMA_VERSION_PATCH 1
+#define LZMA_VERSION_PATCH 2
 
 /**
  * \brief Version stability marker
@@ -32,7 +32,7 @@
  *   - LZMA_VERSION_STABILITY_BETA
  *   - LZMA_VERSION_STABILITY_STABLE
  */
-#define LZMA_VERSION_STABILITY LZMA_VERSION_STABILITY_ALPHA
+#define LZMA_VERSION_STABILITY LZMA_VERSION_STABILITY_BETA
 
 /** \brief Commit version number of the liblzma release */
 #ifndef LZMA_VERSION_COMMIT
diff --git a/src/liblzma/liblzma_generic.map b/src/liblzma/liblzma_generic.map
index cc5f9eba..2b0a1896 100644
--- a/src/liblzma/liblzma_generic.map
+++ b/src/liblzma/liblzma_generic.map
@@ -122,7 +122,7 @@ global:
 	lzma_str_to_filters;
 } XZ_5.2;
 
-XZ_5.5.1alpha {
+XZ_5.5.2beta {
 global:
 	lzma_mt_block_size;
 } XZ_5.4;
diff --git a/src/liblzma/liblzma_linux.map b/src/liblzma/liblzma_linux.map
index b1f0c09b..f0791f2c 100644
--- a/src/liblzma/liblzma_linux.map
+++ b/src/liblzma/liblzma_linux.map
@@ -137,7 +137,7 @@ global:
 	lzma_str_to_filters;
 } XZ_5.2;
 
-XZ_5.5.1alpha {
+XZ_5.5.2beta {
 global:
 	lzma_mt_block_size;
 } XZ_5.4;