commit 85ea0979adcf808a3830aefbe7a4ec884e542ea1
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Feb 3 00:10:34 2023 +0800

    liblzma: Clarify comment in block.h.

diff --git a/src/liblzma/api/lzma/block.h b/src/liblzma/api/lzma/block.h
index b71b55b4..b4eff8b1 100644
--- a/src/liblzma/api/lzma/block.h
+++ b/src/liblzma/api/lzma/block.h
@@ -324,7 +324,8 @@ typedef struct {
  * Note that if the first byte is 0x00, it indicates beginning of Index; use
  * this macro only when the byte is not 0x00.
  *
- * There is no encoding macro, because Block Header encoder is enough for that.
+ * There is no encoding macro, because lzma_block_header_size() and
+ * lzma_block_header_encode() should be used.
  */
 #define lzma_block_header_size_decode(b) (((uint32_t)(b) + 1) * 4)
 