commit 4a972a8ee3ed88ac14067c1d2f15b78988e5dae8
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Dec 6 18:39:03 2023 +0800

    Update THANKS.

diff --git a/THANKS b/THANKS
index 2a5aa28a..c5a0c8aa 100644
--- a/THANKS
+++ b/THANKS
@@ -155,6 +155,7 @@ has been important. :-) In alphabetical order:
   - Mohammed Adnène Trojette
   - Alexey Tourbin
   - Taiki Tsunekawa
+  - Maksym Vatsyk
   - Loganaden Velvindron
   - Patrick J. Volkerding
   - Martin Väth