commit 1e3eb61815a91c0a1bfbb802e2d95593f523491f
Author: jiat75 <jiat0218@gmail.com>
Date:   Fri Jun 3 21:24:54 2022 +0800

    Added parallel test artifacts to .gitignore

diff --git a/.gitignore b/.gitignore
index ca5f024e..62b1d5c7 100644
--- a/.gitignore
+++ b/.gitignore
@@ -43,6 +43,8 @@ build-aux/test-driver
 /src/scripts/xzless
 /src/scripts/xzmore
 
+/tests/*.log
+/tests/*.trs
 /tests/compress_generated_abc
 /tests/compress_generated_random
 /tests/compress_generated_text