commit e13b7947b92355c334edd594295d3a2c99c4bca1
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Oct 18 01:23:26 2023 +0800

    liblzma: Add include guards to crc_common.h.

diff --git a/src/liblzma/check/crc_common.h b/src/liblzma/check/crc_common.h
index 7aef383c..37535f4b 100644
--- a/src/liblzma/check/crc_common.h
+++ b/src/liblzma/check/crc_common.h
@@ -13,6 +13,9 @@
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+#ifndef LZMA_CRC_COMMON_H
+#define LZMA_CRC_COMMON_H
+
 #ifdef WORDS_BIGENDIAN
 #	define A(x) ((x) >> 24)
 #	define B(x) (((x) >> 16) & 0xFF)
@@ -104,3 +107,5 @@ extern uint32_t lzma_crc32_clmul(const uint8_t *buf, size_t size,
 /// CRC64 implemented with the x86 CLMUL instruction.
 extern uint64_t lzma_crc64_clmul(const uint8_t *buf, size_t size,
 		uint64_t crc);
+
+#endif