commit 685094b8e1c1aa1bf934de0366ca42ef599d25f7
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Nov 30 23:10:43 2023 +0800

    Update THANKS.

diff --git a/THANKS b/THANKS
index 0711c53d..2a5aa28a 100644
--- a/THANKS
+++ b/THANKS
@@ -5,6 +5,7 @@ Thanks
 Some people have helped more, some less, but nevertheless everyone's help
 has been important. :-) In alphabetical order:
   - Mark Adler
+  - Kian-Meng Ang
   - H. Peter Anvin
   - Jeff Bastian
   - Nelson H. F. Beebe