commit f27c173d17dc807733b3a4f8c11207c3f04ff34f
Author: jiat75 <jiat0218@gmail.com>
Date:   Sat Oct 30 11:47:15 2021 +0800

    Added error message when archive extraction fails

diff --git a/tar/read.c b/tar/read.c
index 658c810f..b9599661 100644
--- a/tar/read.c
+++ b/tar/read.c
@@ -371,10 +371,9 @@ read_archive(struct bsdtar *bsdtar, char mode, struct archive *writer)
 				r = archive_read_extract2(a, entry, writer);
 			if (r != ARCHIVE_OK) {
 				if (!bsdtar->verbose)
-					safe_fprintf(stderr, "%s",
-					    archive_entry_pathname(entry));
-				safe_fprintf(stderr, ": %s",
-				    archive_error_string(a));
+					safe_fprintf(stderr, "%s", archive_entry_pathname(entry));
+				fprintf(stderr, ": %s: ", archive_error_string(a));
+				fprintf(stderr, "%s", strerror(errno));
 				if (!bsdtar->verbose)
 					fprintf(stderr, "\n");
 				bsdtar->return_value = 1;