commit c247d06e1f6cada9a76f4f6225cbd97ea760f52f
Author: Jia Tan <jiat0218@gmail.com>
Date:   Tue May 2 20:39:56 2023 +0800

    Add NEWS for 5.4.3.

diff --git a/NEWS b/NEWS
index 11bf770d..2f4d35ad 100644
--- a/NEWS
+++ b/NEWS
@@ -2,6 +2,16 @@
 XZ Utils Release Notes
 ======================
 
+5.4.3 (2023-05-04)
+
+    * All fixes from 5.2.12
+
+    * Features in the CMake build can now be disabled as CMake cache
+      variables, similar to the Autotools build.
+
+    * Minor update to the Croatian translation.
+
+
 5.4.2 (2023-03-18)
 
     * All fixes from 5.2.11 that were not included in 5.4.1.