commit 862dacef1a4e7e1b28d465956fa4244ed01df154
Author: Jia Tan <jiat0218@gmail.com>
Date:   Tue Feb 14 00:12:34 2023 +0800

    liblzma: Add one more missing [out] annotation in vli.h

diff --git a/src/liblzma/api/lzma/vli.h b/src/liblzma/api/lzma/vli.h
index bef73267..097801a4 100644
--- a/src/liblzma/api/lzma/vli.h
+++ b/src/liblzma/api/lzma/vli.h
@@ -88,7 +88,7 @@ typedef uint64_t lzma_vli;
  * integer has been encoded.
  *
  * \param       vli       Integer to be encoded
- * \param       vli_pos   How many VLI-encoded bytes have already been written
+ * \param[out]  vli_pos   How many VLI-encoded bytes have already been written
  *                        out. When starting to encode a new integer in
  *                        multi-call mode, *vli_pos must be set to zero.
  *                        To use single-call encoding, set vli_pos to NULL.