commit ba3e4ba2de034ae93a513f9c3a0823b80cdb66dc
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Sep 8 15:07:00 2022 +0300

    CMake: Clarify a comment about Windows symlinks without file extension.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ac8ecbc..7c82f0a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -714,10 +714,9 @@ if(NOT MSVC AND HAVE_GETOPT_LONG)
 
         # Create symlinks in the build directory and then install them.
         #
-        # FIXME? On OSes where executables have a suffix like .exe, this
-        # will create links like unxz -> xz.exe which is correct on Cygwin
-        # but perhaps on some other cases unxz.suffix -> xz.suffix would
-        # be the corrent thing?
+        # The symlinks do not likely need any special extension since
+        # even on Windows the symlink can still be executed without
+        # the .exe extension.
         foreach(LINK IN LISTS XZ_LINKS)
             add_custom_target("${LINK}" ALL
                 "${CMAKE_COMMAND}" -E create_symlink