commit a6583726e5f950278f96abcf79c04f1056810be6
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Jan 6 00:03:35 2023 +0800

    Tests: Use new --filters option in test_compress.sh

diff --git a/tests/test_compress.sh b/tests/test_compress.sh
index 0aa81001..46244cc6 100755
--- a/tests/test_compress.sh
+++ b/tests/test_compress.sh
@@ -133,17 +133,17 @@ test_filter()
 	grep "define HAVE_ENCODER_$1 1" ../config.h > /dev/null || return
 	grep "define HAVE_DECODER_$1 1" ../config.h > /dev/null || return
 	shift
-	test_xz "$@" --lzma2=dict=64KiB,nice=32,mode=fast
+	test_xz --filters="$* lzma2:dict=64KiB,nice=32,mode=fast"
 }
 
-test_filter DELTA --delta=dist=1
-test_filter DELTA --delta=dist=4
-test_filter DELTA --delta=dist=256
-test_filter X86 --x86
-test_filter POWERPC --power
-test_filter IA64 --ia64
-test_filter ARM --arm
-test_filter ARMTHUMB --armthumb
-test_filter SPARC --sparc
+test_filter DELTA delta:dist=1
+test_filter DELTA delta:dist=4
+test_filter DELTA delta:dist=256
+test_filter X86 x86
+test_filter POWERPC powerpc
+test_filter IA64 ia64
+test_filter ARM arm
+test_filter ARMTHUMB armthumb
+test_filter SPARC sparc
 
 exit 0