commit 47a63cad2aa778280e0c1926b7159427ea028cb1
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Apr 21 19:50:14 2023 +0800

    xz: Update --long-help for the new --filtersX option.

diff --git a/src/xz/message.c b/src/xz/message.c
index b9fdc510..b28bb63c 100644
--- a/src/xz/message.c
+++ b/src/xz/message.c
@@ -1030,9 +1030,11 @@ message_help(bool long_help)
 "                      start a new .xz block after every SIZE bytes of input;\n"
 "                      use this to set the block size for threaded compression"));
 		puts(_(
-"      --block-list=SIZES\n"
+"      --block-list=BLOCKS\n"
 "                      start a new .xz block after the given comma-separated\n"
-"                      intervals of uncompressed data"));
+"                      intervals of uncompressed data; optionally, specify a\n"
+"                      filter chain number (0-9) followed by a `:' before the\n"
+"                      uncompressed data size"));
 		puts(_(
 "      --flush-timeout=TIMEOUT\n"
 "                      when compressing, if more than TIMEOUT milliseconds has\n"
@@ -1063,6 +1065,12 @@ message_help(bool long_help)
 "                      syntax; use --filters-help for more information"
 		));
 
+		puts(_(
+"  --filters1=FILTERS ... --filters9=FILTERS\n"
+"                      set additional filter chains using the liblzma filter\n"
+"                      string syntax to use with --block-list"
+		));
+
 #if defined(HAVE_ENCODER_LZMA1) || defined(HAVE_DECODER_LZMA1) \
 		|| defined(HAVE_ENCODER_LZMA2) || defined(HAVE_DECODER_LZMA2)
 		// TRANSLATORS: The word "literal" in "literal context bits"