commit 78e0561dfebaa9d5e34558de537efcda890e0629
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Jan 6 20:43:31 2023 +0800

    Style: Change #if !defined() to #ifndef in mythread.h.

diff --git a/src/common/mythread.h b/src/common/mythread.h
index a0dced19..37b5216b 100644
--- a/src/common/mythread.h
+++ b/src/common/mythread.h
@@ -108,7 +108,7 @@ mythread_sigmask(int how, const sigset_t *restrict set,
 // If clock_gettime() isn't available, use gettimeofday() from <sys/time.h>
 // as a fallback. gettimeofday() is in SUSv2 and thus is supported on all
 // relevant POSIX systems.
-#if !defined(HAVE_CLOCK_GETTIME)
+#ifndef HAVE_CLOCK_GETTIME
 #	include <sys/time.h>
 #endif
 