commit c9c8bfae3502842dcead85eeb2b951b437c2cd88
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Feb 22 20:59:41 2023 +0800

    CMake: Add LZIP decoder test to list of tests.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e4ae1ec..44869e59 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -852,6 +852,7 @@ if(BUILD_TESTING)
         test_hardware
         test_index
         test_index_hash
+        test_lzip_decoder
         test_memlimit
         test_stream_flags
         test_vli