commit 064cd385a716abc78d93a3612411a82d69ceb221
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Dec 29 00:30:52 2022 +0800

    Adds test_index_hash to .gitignore.

diff --git a/.gitignore b/.gitignore
index 44f269cc..7f2dec23 100644
--- a/.gitignore
+++ b/.gitignore
@@ -62,6 +62,7 @@ coverage
 /tests/test_filter_flags
 /tests/test_hardware
 /tests/test_index
+/tests/test_index_hash
 /test/test_lzip_decoder
 /tests/test_memlimit
 /tests/test_stream_flags