commit 6bcd516812331de42b347922913230895bebad34
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Jun 7 00:10:38 2023 +0800

    Update THANKS.

diff --git a/THANKS b/THANKS
index cad65242..03ec8e09 100644
--- a/THANKS
+++ b/THANKS
@@ -19,6 +19,7 @@ has been important. :-) In alphabetical order:
   - Jakub Bogusz
   - Adam Borowski
   - Maarten Bosmans
+  - Benjamin Buch
   - Trent W. Buck
   - Kevin R. Bulgrien
   - James Buren