commit a5c177f514f4c90e0d2f6045636fca6c2e80a20d
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Feb 2 01:39:28 2024 +0800

    Update AUTHORS.

diff --git a/AUTHORS b/AUTHORS
index 69bbfc3e..01730ed7 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -1,39 +1,40 @@
 
 Authors of XZ Utils
 ===================
 
     XZ Utils is developed and maintained by Lasse Collin
     <lasse.collin@tukaani.org> and Jia Tan <jiat0218@gmail.com>.
 
     Major parts of liblzma are based on code written by Igor Pavlov,
     specifically the LZMA SDK <https://7-zip.org/sdk.html>. Without
     this code, XZ Utils wouldn't exist.
 
     The SHA-256 implementation in liblzma is based on the code found from
     7-Zip <https://7-zip.org/>, which has a modified version of the SHA-256
     code found from Crypto++ <https://www.cryptopp.com/>. The SHA-256 code
     in Crypto++ was written by Kevin Springle and Wei Dai.
 
     Some scripts have been adapted from gzip. The original versions
     were written by Jean-loup Gailly, Charles Levert, and Paul Eggert.
     Andrew Dudman helped adapting the scripts and their man pages for
     XZ Utils.
 
     The initial version of the threaded .xz decompressor was written
     by Sebastian Andrzej Siewior.
 
     The initial version of the .lz (lzip) decoder was written
     by Michał Górny.
 
-    CLMUL-accelerated CRC code was contributed by Ilya Kurdyukov.
+    Architecture-specific CRC optimizations were contributed by
+    Ilya Kurdyukov, Hans Jansen, and Chenxi Mao.
 
     Other authors:
       - Jonathan Nieder
       - Joachim Henke
 
     The GNU Autotools-based build system contains files from many authors,
     which I'm not trying to list here.
 
     Several people have contributed fixes or reported bugs. Most of them
     are mentioned in the file THANKS.
 