commit adb073da76a920b5a81e6b32254f4ddb054dc57a
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Feb 9 23:59:54 2024 +0800

    liblzma: Fix typo discovered by codespell.

diff --git a/src/liblzma/check/crc32_arm64.h b/src/liblzma/check/crc32_arm64.h
index a9722bdb..fc24cd82 100644
--- a/src/liblzma/check/crc32_arm64.h
+++ b/src/liblzma/check/crc32_arm64.h
@@ -61,7 +61,7 @@ crc32_arch_optimized(const uint8_t *buf, size_t size, uint32_t crc)
 
 	// Process 8 bytes at a time. The end point is determined by
 	// ignoring the least significant three bits of size to ensure
-	// we do not process past the bounds of the buffer. This guarentees
+	// we do not process past the bounds of the buffer. This guarantees
 	// that limit is a multiple of 8 and is strictly less than size.
 	for (const uint8_t *limit = buf + (size & ~((size_t)7));
 			buf < limit; buf += 8)