commit 4110a998b83459fe2bc9bc1bec30ad68afa8f797
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Jan 12 23:09:03 2023 +0800

    CI: Allow disabling Native Language Support.

diff --git a/build-aux/ci_build.sh b/build-aux/ci_build.sh
index 0e43d29f..e2c09cdb 100755
--- a/build-aux/ci_build.sh
+++ b/build-aux/ci_build.sh
@@ -19,7 +19,7 @@ set -e
 USAGE="Usage: $0
   -b [autotools|cmake]
   -c [crc32|crc64|sha256]
-  -d [encoders|decoders|bcj|delta|threads|shared]
+  -d [encoders|decoders|bcj|delta|threads|shared|nls]
   -l [destdir]
   -s [srcdir]
   -p [all|build|test]
@@ -37,6 +37,7 @@ ENCODERS="y"
 DECODERS="y"
 THREADS="y"
 SHARED="y"
+NATIVE_LANG_SUPPORT="y"
 SRC_DIR="$ABS_DIR/../"
 DEST_DIR="$SRC_DIR/../xz_build"
 PHASE="all"
@@ -80,6 +81,7 @@ while getopts b:c:d:l:s:p:f:h opt; do
 		delta) DELTA="n" ;;
 		threads) THREADS="n" ;;
 		shared) SHARED="n";;
+		nls) NATIVE_LANG_SUPPORT="n";;
 		*) echo "Invalid disable value: $disable_arg"; exit 1 ;;
 		esac
 	done	
@@ -150,6 +152,11 @@ if [ "$PHASE" = "all" ] || [ "$PHASE" = "build" ]; then
 			EXTRA_OPTIONS="$EXTRA_OPTIONS --disable-shared"
 		fi
 
+		if [ "$NATIVE_LANG_SUPPORT" = "n" ]
+		then
+			EXTRA_OPTIONS="$EXTRA_OPTIONS --disable-nls"
+		fi
+
 		# Run configure script
 		"$SRC_DIR"/configure --enable-werror --enable-checks=$CHECK_TYPE $EXTRA_OPTIONS
 