commit a5de68bac2bb7e1b9119e6cea7d761a22ea73e9c
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Feb 17 00:44:44 2023 +0800

    liblzma: Adjust documentation in bcj.h for consistent style.

diff --git a/src/liblzma/api/lzma/bcj.h b/src/liblzma/api/lzma/bcj.h
index 120a0003..45a0fc63 100644
--- a/src/liblzma/api/lzma/bcj.h
+++ b/src/liblzma/api/lzma/bcj.h
@@ -20,40 +20,41 @@
 
 /* Filter IDs for lzma_filter.id */
 
+/**
+ * \brief Filter for x86 binaries
+ */
 #define LZMA_FILTER_X86         LZMA_VLI_C(0x04)
-	/**<
-	 * Filter for x86 binaries
-	 */
 
+/**
+ * \brief Filter for Big endian PowerPC binaries
+ */
 #define LZMA_FILTER_POWERPC     LZMA_VLI_C(0x05)
-	/**<
-	 * Filter for Big endian PowerPC binaries
-	 */
 
+/**
+ * \brief Filter for IA-64 (Itanium) binaries
+ */
 #define LZMA_FILTER_IA64        LZMA_VLI_C(0x06)
-	/**<
-	 * Filter for IA-64 (Itanium) binaries.
-	 */
 
+/**
+ * \brief Filter for ARM binaries
+ */
 #define LZMA_FILTER_ARM         LZMA_VLI_C(0x07)
-	/**<
-	 * Filter for ARM binaries.
-	 */
 
+/**
+ * \brief Filter for ARM-Thumb binaries
+ */
 #define LZMA_FILTER_ARMTHUMB    LZMA_VLI_C(0x08)
-	/**<
-	 * Filter for ARM-Thumb binaries.
-	 */
 
+/**
+ * \brief Filter for SPARC binaries
+ */
 #define LZMA_FILTER_SPARC       LZMA_VLI_C(0x09)
-	/**<
-	 * Filter for SPARC binaries.
-	 */
 
+/**
+ * \brief Filter for ARM64 binaries
+ */
 #define LZMA_FILTER_ARM64       LZMA_VLI_C(0x0A)
-	/**<
-	 * Filter for ARM64 binaries.
-	 */
+
 
 /**
  * \brief       Options for BCJ filters (except ARM64)