commit 2959dbc7358efcf421ce51bc9cd7eae8fdd8fec4
Author: Jia Tan <jiat0218@gmail.com>
Date:   Mon Jan 22 23:33:39 2024 +0800

    liblzma: Update string_conversion.c to support RISC-V Filter.

diff --git a/src/liblzma/common/string_conversion.c b/src/liblzma/common/string_conversion.c
index d2c1e809..ad8241e2 100644
--- a/src/liblzma/common/string_conversion.c
+++ b/src/liblzma/common/string_conversion.c
@@ -509,6 +509,11 @@ static const struct {
 	  &parse_bcj,     bcj_optmap, 1, 1, true },
 #endif
 
+#if defined(HAVE_ENCODER_RISCV) || defined(HAVE_DECODER_RISCV)
+	{ "riscv",        sizeof(lzma_options_bcj),   LZMA_FILTER_RISCV,
+	  &parse_bcj,     bcj_optmap, 1, 1, true },
+#endif
+
 #if defined(HAVE_ENCODER_POWERPC) || defined(HAVE_DECODER_POWERPC)
 	{ "powerpc",      sizeof(lzma_options_bcj),   LZMA_FILTER_POWERPC,
 	  &parse_bcj,     bcj_optmap, 1, 1, true },