commit d9acf3977948dbb3bf27b0f70b0ac476fc7ca448
Author: jiat75 <jiat0218@gmail.com>
Date:   Thu Oct 7 22:43:08 2021 +0800

    Removed unused function.

diff --git a/libarchive/test/test_archive_write_add_filter_by_name.c b/libarchive/test/test_archive_write_add_filter_by_name.c
index b9173984..49f91ac5 100644
--- a/libarchive/test/test_archive_write_add_filter_by_name.c
+++ b/libarchive/test/test_archive_write_add_filter_by_name.c
@@ -142,12 +142,6 @@ canAlways(void)
 	return 1;
 }
 
-static int
-cannot(void)
-{
-	return 0;
-}
-
 DEFINE_TEST(test_archive_write_add_filter_by_name_b64encode)
 {
 	test_filter_by_name("b64encode", ARCHIVE_FILTER_UU, canAlways);