commit 116e81f002c503d3c3cd12726db8f9116e58ef25
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Mar 22 15:42:04 2023 +0800

    Build: Removes redundant check for LZMA1 filter support.

diff --git a/src/liblzma/lzma/Makefile.inc b/src/liblzma/lzma/Makefile.inc
index 506b328c..25440d85 100644
--- a/src/liblzma/lzma/Makefile.inc
+++ b/src/liblzma/lzma/Makefile.inc
@@ -7,12 +7,9 @@
 
 EXTRA_DIST += lzma/fastpos_tablegen.c
 
-liblzma_la_SOURCES += lzma/lzma_common.h
-
-if COND_FILTER_LZMA1
 liblzma_la_SOURCES += \
+	lzma/lzma_common.h \
 	lzma/lzma_encoder_presets.c
-endif
 
 if COND_ENCODER_LZMA1
 liblzma_la_SOURCES += \