commit 504cf4af895fd45aad0c56eb3b49d90acd54465b
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Feb 3 00:11:37 2023 +0800

    liblzma: Minor edits to lzma_block header_size documentation.

diff --git a/src/liblzma/api/lzma/block.h b/src/liblzma/api/lzma/block.h
index 6522a23e..71a0513d 100644
--- a/src/liblzma/api/lzma/block.h
+++ b/src/liblzma/api/lzma/block.h
@@ -62,7 +62,7 @@ typedef struct {
 	uint32_t version;
 
 	/**
-	 * \brief       Size of the Block Header field
+	 * \brief       Size of the Block Header field in bytes
 	 *
 	 * This is always a multiple of four.
 	 *
@@ -78,6 +78,7 @@ typedef struct {
 	 * Written by:
 	 *  - lzma_block_header_size()
 	 *  - lzma_block_buffer_encode()
+	 *  - lzma_block_uncomp_encode()
 	 */
 	uint32_t header_size;
 #	define LZMA_BLOCK_HEADER_SIZE_MIN 8