commit 115b720fb521f99aa832d06b2c12b7f8c6c50680
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Feb 3 00:11:07 2023 +0800

    liblzma: Enumerate functions that read version in lzma_block.

diff --git a/src/liblzma/api/lzma/block.h b/src/liblzma/api/lzma/block.h
index b4eff8b1..6522a23e 100644
--- a/src/liblzma/api/lzma/block.h
+++ b/src/liblzma/api/lzma/block.h
@@ -44,8 +44,17 @@ typedef struct {
 	 * with any version value).
 	 *
 	 * Read by:
-	 *  - All functions that take pointer to lzma_block as argument,
-	 *    including lzma_block_header_decode().
+	 *  - lzma_block_header_size()
+	 *  - lzma_block_header_encode()
+	 *  - lzma_block_header_decode()
+	 *  - lzma_block_compressed_size()
+	 *  - lzma_block_unpadded_size()
+	 *  - lzma_block_total_size()
+	 *  - lzma_block_encoder()
+	 *  - lzma_block_decoder()
+	 *  - lzma_block_buffer_encode()
+	 *  - lzma_block_uncomp_encode()
+	 *  - lzma_block_buffer_decode()
 	 *
 	 * Written by:
 	 *  - lzma_block_header_decode()