commit ff7badef53c2cd698d4b72b945f34dfd0835e13c
Author: Jia Tan <jiat0218@gmail.com>
Date:   Fri Nov 24 21:19:12 2023 +0800

    CMake: Use consistent indentation with check_c_source_compiles().

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b1753c37..1e9af0e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -909,8 +909,8 @@ calculation if supported by the system" ON)
                     return _mm_clmulepi64_si128(a, b, 0);
                 }
                 int main(void) { return 0; }
-        "
-        HAVE_USABLE_CLMUL)
+            "
+            HAVE_USABLE_CLMUL)
 
         if(HAVE_USABLE_CLMUL)
             target_sources(liblzma PRIVATE src/liblzma/check/crc_clmul.c)