commit 77050b78364ffb6b0f129e742b7c31602d725c08
Author: Jia Tan <jiat0218@gmail.com>
Date:   Tue May 2 20:39:37 2023 +0800

    Add NEWS for 5.2.12.

diff --git a/NEWS b/NEWS
index 47a8a7cd..11bf770d 100644
--- a/NEWS
+++ b/NEWS
@@ -677,6 +677,20 @@ XZ Utils Release Notes
       (FreeBSD >= 10).
 
 
+5.2.12 (2023-05-04)
+
+    * Fixed a build system bug that prevented building liblzma as a
+      shared library when configured with --disable-threads. This bug
+      affected releases 5.2.6 to 5.2.11 and 5.4.0 to 5.4.2.
+
+    * Include <intrin.h> for Windows intrinsic functions where they are
+      needed. This fixed a bug that prevented building liblzma using
+      clang-cl on Windows.
+
+    * Minor update to the Croatian translation. The small change
+      applies to a string in both 5.2 and 5.4 branches.
+
+
 5.2.11 (2023-03-18)
 
     * Removed all possible cases of null pointer + 0. It is undefined