commit 30a25f3742287697bc57a1bef86c19ecf5129322
Author: Jia Tan <jiat0218@gmail.com>
Date:   Mon Jan 22 22:08:45 2024 +0800

    Docs: Add --disable-arm64-crc32 description to INSTALL.

diff --git a/INSTALL b/INSTALL
index d6eaa616..670e3989 100644
--- a/INSTALL
+++ b/INSTALL
@@ -372,7 +372,7 @@ XZ Utils Installation
                 code.
 
     --disable-clmul-crc
-                Disable the use carryless multiplication for CRC
+                Disable the use of carryless multiplication for CRC
                 calculation even if compiler support for it is detected.
                 The code uses runtime detection of SSSE3, SSE4.1, and
                 CLMUL instructions on x86. On 32-bit x86 this currently
@@ -383,6 +383,16 @@ XZ Utils Installation
                 required extensions (-msse4.1 -mpclmul) then runtime
                 detection isn't used and the generic code is omitted.
 
+    --disable-arm64-crc32
+                Disable the use of the ARM64 CRC32 instruction extension
+                even if compiler support for it is detected. The code will
+                detect support for the instruction at runtime.
+
+                If using compiler options that unconditionally allow the
+                required extensions (-march=armv8-a+crc or -march=armv8.1-a
+                and later) then runtime detection isn't used and the
+                generic code is omitted.
+
     --enable-unaligned-access
                 Allow liblzma to use unaligned memory access for 16-bit,
                 32-bit, and 64-bit loads and stores. This should be