commit 54d2b66049cd25d60f84047c454b920702918cae
Author: jiat75 <jiat0218@gmail.com>
Date:   Sat Oct 9 13:58:44 2021 +0800

    Added bin folder to .gitignore

diff --git a/.gitignore b/.gitignore
index 9c09bf49..6b4d2dc7 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,73 +1,74 @@
 *~
 *.o
 *.exe
 *.la
 *.lo
 *.cmake
 !build/cmake/*.cmake
 .deps/
 .libs/
+bin/
 .dirstamp
 Makefile
 Makefile.in
 aclocal.m4
 autom4te.cache/
 bsdcat
 bsdcpio
 bsdtar
 build/autoconf/compile
 build/autoconf/config.guess
 build/autoconf/config.sub
 build/autoconf/depcomp
 build/autoconf/install-sh
 build/autoconf/libtool.m4
 build/autoconf/ltmain.sh
 build/autoconf/ltoptions.m4
 build/autoconf/ltsugar.m4
 build/autoconf/ltversion.m4
 build/autoconf/lt~obsolete.m4
 build/autoconf/missing
 build/pkgconfig/libarchive.pc
 cat/test/list.h
 config.cache
 config.h
 config.h.in
 config.log
 config.status
 configure
 cpio/test/list.h
 libarchive/test/list.h
 libtool
 stamp-h1
 tar/test/list.h
 CMakeCache.txt
 CMakeFiles/
 DartConfiguration.tcl
 cmake.tmp/
 
 doc/html/*.html
 doc/man/*.1
 doc/man/*.3
 doc/man/*.5
 doc/pdf/*.pdf
 doc/text/*.txt
 doc/wiki/*.wiki
 
 libarchive-*.tar.gz
 libarchive-*.zip
 
 Testing/
 libarchive/libarchive.a
 libarchive/libarchive.so
 libarchive/libarchive.so.*
 
 .DS_Store
 
 /*_test
 /*_test.log
 /*_test.trs
 build/autoconf/test-driver
 test-suite.log
 
 .sw?
 .*.sw?