commit e280470040b27c5e58d78b25b9e2bb71fc6c3882
Author: Jia Tan <jiat0218@gmail.com>
Date:   Tue Jan 9 16:56:16 2024 +0800

    Build: Add the logo and license to the release.

diff --git a/Makefile.am b/Makefile.am
index f562f902..d6bfba68 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -61,6 +61,7 @@ endif
 EXTRA_DIST = \
 	po4a \
 	extra \
+	doc/logo \
 	dos \
 	doxygen \
 	windows \
@@ -68,6 +69,7 @@ EXTRA_DIST = \
 	cmake \
 	CMakeLists.txt \
 	autogen.sh \
+	COPYING.CC-BY-SA-4.0 \
 	COPYING.GPLv2 \
 	COPYING.GPLv3 \
 	COPYING.LGPLv2.1 \