commit db2b4aa068a492c0013279a4ed43803e8ff9bb3e
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Nov 22 21:12:15 2023 +0800

    CI: Allow ci_build.sh to set a different C compiler.

diff --git a/build-aux/ci_build.sh b/build-aux/ci_build.sh
index 41cc2bc5..51b611a6 100755
--- a/build-aux/ci_build.sh
+++ b/build-aux/ci_build.sh
@@ -23,6 +23,7 @@ USAGE="Usage: $0
   -d [encoders|decoders|bcj|delta|threads|shared|nls|small|ifunc|clmul|sandbox]
   -f [CFLAGS]
   -l [destdir]
+  -m [compiler]
   -n [ARTIFACTS_DIR_NAME]
   -p [all|build|test]
   -s [srcdir]"
@@ -55,7 +56,7 @@ ARTIFACTS_DIR_NAME="output"
 # Parse arguments #
 ###################
 
-while getopts a:b:c:d:l:n:s:p:f:h opt; do
+while getopts a:b:c:d:l:m:n:s:p:f:h opt; do
 	# b option can have either value "autotools" OR "cmake"
 	case ${opt} in
 	h)
@@ -97,6 +98,10 @@ while getopts a:b:c:d:l:n:s:p:f:h opt; do
 	;;
 	l) DEST_DIR="$OPTARG"
 	;;
+	m)
+		CC="$OPTARG"
+		export CC
+	;;
 	n) ARTIFACTS_DIR_NAME="$OPTARG"
 	;;
 	s) SRC_DIR="$OPTARG"
@@ -237,6 +242,10 @@ then
 
 		add_extra_option "$SMALL" "-DHAVE_SMALL=ON" ""
 
+		if test -n "$CC" ; then
+			EXTRA_OPTIONS="$EXTRA_OPTIONS -DCMAKE_C_COMPILER=$CC"
+		fi
+
 		# Remove old cache file to clear previous settings.
 		rm -f "CMakeCache.txt"
 		cmake "$SRC_DIR/CMakeLists.txt" -B "$DEST_DIR" $EXTRA_OPTIONS -DADDITIONAL_CHECK_TYPES="$CHECK_TYPE" -G "Unix Makefiles"