commit af4925e6043113ec9b5f9c0cf13abf2a18ccb1f6
Author: Jia Tan <jiat0218@gmail.com>
Date:   Sat Mar 18 22:10:12 2023 +0800

    Add NEWS for 5.2.11.

diff --git a/NEWS b/NEWS
index 88aa55ac..7144bcee 100644
--- a/NEWS
+++ b/NEWS
@@ -629,6 +629,33 @@ XZ Utils Release Notes
       (FreeBSD >= 10).
 
 
+5.2.11 (2023-03-18)
+
+    * Removed all possible cases of null pointer + 0. It is undefined
+      behavior in C99 and C17. This was detected by a sanitizer and had
+      not caused any known issues.
+
+    * Build systems:
+
+        - Added a workaround for building with GCC on MicroBlaze Linux.
+          GCC 12 on MicroBlaze doesn't support the __symver__ attribute
+          even though __has_attribute(__symver__) returns true. The
+          build is now done without the extra RHEL/CentOS 7 symbols
+          that were added in XZ Utils 5.2.7. The workaround only
+          applies to the Autotools build (not CMake).
+
+        - CMake: Ensure that the C compiler language is set to C99 or
+          a newer standard.
+
+        - CMake changes from XZ Utils 5.4.1:
+
+            * Added a workaround for a build failure with
+              windres from GNU binutils.
+
+            * Included the Windows resource files in the xz
+              and xzdec build rules.
+
+
 5.2.10 (2022-12-13)
 
     * xz: Don't modify argv[] when parsing the --memlimit* and