commit 148e20607e95781558bdfc823ecba07b7af4b590
Author: Jia Tan <jiat0218@gmail.com>
Date:   Tue Oct 31 21:51:40 2023 +0800

    Build: Fix text wrapping in an output message.

diff --git a/configure.ac b/configure.ac
index 553a1b87..1fe61016 100644
--- a/configure.ac
+++ b/configure.ac
@@ -857,11 +857,12 @@ CFLAGS="$OLD_CFLAGS"
 # win95, and the compiler does not support attribute constructor, then we
 # would end up with a multithreaded build that is thread-unsafe. As a
 # result this configuration is not allowed.
-if test "x$enable_small$enable_threads$have_func_attribute_constructor"\
+if test "x$enable_small$enable_threads$have_func_attribute_constructor" \
 		= xyeswin95no; then
-	AC_MSG_ERROR([--enable-threads=win95 and --enable-small cannot be
-		used at the same time with a compiler that doesn't support
-		__attribute__((__constructor__))])
+	AC_MSG_ERROR([
+    --enable-threads=win95 and --enable-small cannot be used
+    at the same time with a compiler that doesn't support
+    __attribute__((__constructor__))])
 fi
 
 # __attribute__((__ifunc__())) can be used to choose between different